openstack/clustering/v1/policytypes/results.go

```diff
@@ -36,3 +37,68 @@ func (page PolicyTypePage) IsEmpty() (bool, error) {
 	policyTypes, err := ExtractPolicyTypes(page)
 	return len(policyTypes) == 0, err
 }
+
+// PolicyTypeDetail represents the detailed policy type information for a clustering policy type
+type PolicyTypeDetail struct {
+	Name          string                         `json:"name"`
+	Schema        SchemaType                     `json:"schema"`
+	SupportStatus map[string][]SupportStatusType `json:"support_status,omitempty"`
+}
+
+// SchemaType represents the schema of a clustering policy type
+type SchemaType struct {
```

Comment:
For each set of policy type fields, will a result only ever have one set returned or is it possible for a policy type to have values for, for example, affinity, batch, and health?

If a result will only ever have `senlin.policy.affinity` set, then we're leaving a lot of unused fields here. In addition, if Senlin adds more policies types in later releases, then they wouldn't be parsable here until Gophercloud was updated. Finally, how would a user know what fields they should be looking at if the bulk of them are empty?
If this is all true, I wonder if we should just make `Schema` a generic `map[string]interface{}`.

Reply:
Only one set will ever be returned.  I've had the same discussion with @problemv and @jackkuei, and we couldn't come to a clear consensus. Our thinking was that the structs make it easier for a developer using Gophercloud to access the policy type fields.

Usually it is rare that Senlin adds a new policy type.  However, a more frequent case is that the policy types are extended (i.e. from version 1.0 to 1.1) with the addition of new fields.  In that case your concern is valid that an update to Gophercloud is required before the new policy type would be usable.

I also prefer making `Schema` a generic `map[string]interface{}` as long as you are OK with the Gophercloud users having to iterate thru the `map[string]interface{}` to retrieve each field.

Reply:
> Our thinking was that the structs make it easier for a developer using Gophercloud to access the policy type fields.

This is true. However, in this case I think there's too much ambiguity with all of the fields to make this struct easily usable.

If the policy types are standard (meaning they're built-in to Senlin and users can't create their own), one other option is to make distinct structs for each policy schema type and then have individual `ExtractXXXX` functions where XXXX is something like `ExtractAffinityPolicySchema`. 

I think the decision here should be based on how often a user is going to consume a Policy Type Schema. If parsing the policy is an important action in Senlin, then the Extract method is the way to go (but also keep in mind that new fields will be omitted until they are added to the struct). But if looking at the policy schema is something which will only be done in niche cases, using a `map[string]interface{}` should be fine -- at least to start with.

Reply:
Ok, thanks for the clarification.  Inspecting the policy type schema is definitely a rare case, so I will go with the `map[string]interface{}` implementation in this case.